tests/test_spark.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import logging
+from time import sleep
+from typing import List
+
+import pytest
+from docker import DockerClient
+from docker.models.containers import Container
+from docker.types import Mount
+from granulate_utils.metrics import MetricsSnapshot
+from granulate_utils.metrics.metrics import (
+    SPARK_AGGREGATED_STAGE_METRICS,
+    SPARK_APPLICATION_DIFF_METRICS,
+    SPARK_APPLICATION_GAUGE_METRICS,
+    SPARK_EXECUTORS_METRICS,
+)
+from granulate_utils.metrics.sampler import BigDataSampler
+from pytest import LogCaptureFixture
+
+from gprofiler.log import get_logger_adapter
+from tests.conftest import build_image
+from tests.utils import wait_container_to_start
+
+# `BigDataSampler` receives a logger as an argument.
+logger = get_logger_adapter("gprofiler_spark_test")
+
+# List that includes all the metrics that are expected to be collected by `BigDataSampler` in SA mode.
+EXPECTED_SA_METRICS_KEYS = [
+    metric
+    for metrics_dict in (
+        SPARK_APPLICATION_GAUGE_METRICS,
+        SPARK_APPLICATION_DIFF_METRICS,
+        SPARK_AGGREGATED_STAGE_METRICS,
+        SPARK_EXECUTORS_METRICS,
+    )
+    for metric in metrics_dict.values()
+]
+
+SPARK_MASTER_HOST = "127.0.0.1"
+
+DISCOVER_INTERVAL_SECS = 5
+DISCOVER_TIMEOUT_SECS = 60
+
+
+@pytest.fixture
+def sparkpi_container(docker_client: DockerClient, application_docker_mounts: List[Mount]) -> Container:
+    """
+    This fixture is responsible for running SparkPi application in a container.
+    See `containers/spark/Dockerfile`
+    """
+    spark_image = build_image(docker_client=docker_client, runtime="spark")
+    container = docker_client.containers.run(
+        spark_image,
+        detach=True,
+        mounts=application_docker_mounts,
+        network_mode="host",
+        pid_mode="host",
+        environment={"SPARK_MASTER_HOST": SPARK_MASTER_HOST},
+    )
+    wait_container_to_start(container)
+    try:
+        yield container
+    finally:
+        container.stop()
+        container.remove()
+
+
+def _validate_sa_metricssnapshot(snapshot: MetricsSnapshot) -> None:
+    """
+    Validates that the snapshot contains all the expected metrics.
+    """
+    samples = snapshot.samples
+    assert len(samples) != 0, "No samples found in snapshot"
+    metric_keys = [sample.name for sample in samples]
+    for key in EXPECTED_SA_METRICS_KEYS:
+        assert key in metric_keys, f"Metric {key} not found in snapshot"
+
+
+def test_sa_spark_discovered_mode(caplog: LogCaptureFixture, sparkpi_container: Container) -> None:
```

Comment:
Test names - use `test_spark_XXX` (as "spark" is the category here, not "sa"). Other tests follow this pattern, e.g `test_java`, `test_python` etc.

Reply:
68b77d3